### PR TITLE
Fix ipv6 address parsing issue

### PR DESCRIPTION
**Pull request by sreecha**
This was found by one of the envoy fuzz tests.

The parsing logic fails if the IPV6 address is longer than `INET6_ADDRSTRLEN`

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,955,154      Total (>)      1,955,026

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,843,754      Total (<)     10,843,756

 No significant differences in binary sizes


```

**Comment by jtattermusch** [comment]
Please add a PR description.

File: src/core/ext/filters/client_channel/parse_address.cc [hunk 1]
```diff
@@ -128,6 +128,13 @@ bool grpc_parse_ipv6_hostport(const char* hostport, grpc_resolved_address* addr,
     char host_without_scope[GRPC_INET6_ADDRSTRLEN];
     size_t host_without_scope_len = static_cast<size_t>(host_end - host);
     uint32_t sin6_scope_id = 0;
+    if (host_without_scope_len > GRPC_INET6_ADDRSTRLEN) {
```
Comment (vjpai) [inline src/core/ext/filters/client_channel/parse_address.cc]: >=

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                            FILE SIZE
 ++++++++++++++ GROWING                                              ++++++++++++++
  +0.0%    +112 [None]                                                   +80  +0.0%
  +2.8%     +48 src/core/ext/filters/client_channel/parse_address.cc     +48  +2.8%
      +5.2%     +39 grpc_parse_ipv6_hostport                                 +39  +5.2%
       +30%      +9 [Unmapped]                                                +9   +30%

  +0.0%    +160 TOTAL                                                   +128  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,955,154      Total (>)      1,955,026

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,843,756      Total (<)     10,843,757

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,955,155      Total (>)      1,955,026

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,843,752      Total (>)     10,843,751

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                            FILE SIZE
 ++++++++++++++ GROWING                                              ++++++++++++++
  +0.0%     +80 [None]                                                   +80  +0.0%
  +2.8%     +48 src/core/ext/filters/client_channel/parse_address.cc     +48  +2.8%
      +5.2%     +39 grpc_parse_ipv6_hostport                                 +39  +5.2%
       +30%      +9 [Unmapped]                                                +9   +30%

  +0.0%    +128 TOTAL                                                   +128  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,955,131      Total (>)      1,955,002

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,843,754      Total (<)     10,843,756

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by htuch** [comment]
@sreecha any ETA on when this will be merged and we can bump Envoy to benefit from it?

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                            FILE SIZE
 ++++++++++++++ GROWING                                              ++++++++++++++
  +0.0%     +80 [None]                                                   +72  +0.0%
  +2.8%     +48 src/core/ext/filters/client_channel/parse_address.cc     +48  +2.8%
      +5.2%     +39 grpc_parse_ipv6_hostport                                 +39  +5.2%
       +30%      +9 [Unmapped]                                                +9   +30%

  +0.0%    +128 TOTAL                                                   +120  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,956,097      Total (>)      1,955,968

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,843,886      Total (>)     10,843,885

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                            FILE SIZE
 ++++++++++++++ GROWING                                              ++++++++++++++
  +0.0%     +80 [None]                                                   +72  +0.0%
  +2.8%     +48 src/core/ext/filters/client_channel/parse_address.cc     +48  +2.8%
      +5.2%     +39 grpc_parse_ipv6_hostport                                 +39  +5.2%
       +30%      +9 [Unmapped]                                                +9   +30%

  +0.0%    +128 TOTAL                                                   +120  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,956,097      Total (>)      1,955,968

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,843,886      Total (<)     10,843,888

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Pull request merged**